common/src/main/java/com/wynntils/overlays/StatusEffectsOverlay.java

```diff
@@ -138,52 +139,92 @@ private Stream<RenderedStatusEffect> stackEffects(List<StatusEffect> effects) {
         return effectsToRender.values().stream();
     }
 
-    private static final class RenderedStatusEffect {
+    private String getEffectsKey(StatusEffect effect) {
+        return switch (effectsStackBehaviour.get()) {
+            default -> effect.asString().getString();
+            case SUM -> effect.getPrefix().getString()
+                    + effect.getName().getString()
+                    + effect.getModifier().getString().indexOf('%')
+                    + effect.getDisplayedTime().getString();
+        };
+    }
+
+    private final class RenderedStatusEffect {
         private final StatusEffect effect;
 
         private int count = 0;
+        private List<String> modifierList = new ArrayList<>();
 
         private RenderedStatusEffect(StatusEffect effect) {
             this.effect = effect;
         }
 
         private StyledText getRenderedText() {
-            if (this.count == 1) {
+            if (this.count <= 1) {
+                // Terminate early if there's nothing to do
                 return this.effect.asString();
             }
 
-            String modifierString = this.effect.getModifier().getString();
-            StyledText modifierText;
-
-            // look for either a - or a +
-            int minusIndex = modifierString.indexOf('-');
-            int plusIndex = modifierString.indexOf('+');
-            int index = Math.max(minusIndex, plusIndex);
-
-            if (index == -1) {
-                // We can simply put the count string at the start
-                modifierText = StyledText.fromString(ChatFormatting.GRAY + (this.count + "x"))
-                        .append(this.effect.getModifier());
-            } else {
-                // The count string is inserted between the +/- and the number
-                index += 1;
-                modifierText = StyledText.fromString(ChatFormatting.GRAY
-                        + modifierString.substring(0, index)
-                        + (this.count + "x")
-                        + modifierString.substring(index));
+            StyledText modifierText = StyledText.EMPTY;
+
+            switch (effectsStackBehaviour.get()) {
+                case SUM -> {
+                    if (this.modifierList.size() > 0) {
+                        // SUM modifiers
+                        double modifierValue = 0.0;
+                        String baseModifier = this.modifierList.get(0);
+                        for (String modifier : modifierList) {
+                            modifierValue += extractDoubleFromString(modifier);
+                        }
+
+                        // Eliminate .0 when the modifier needs trailing decimals. This is the case for powder specials
+                        // on armor.
+                        String numberString = (Math.round(modifierValue) == modifierValue)
+                                ? String.format("%+d", (long) modifierValue)
+                                : String.format("%+.1f", modifierValue);
```

Comment:
This format assumes positive effects only?

Reply:
Nope, the + in the formatting will add a + to the numbers when they are positive.